explorer/src/providers/accounts/history.tsx

```diff
@@ -4,19 +4,26 @@ import {
   ConfirmedSignatureInfo,
   TransactionSignature,
   Connection,
+  ParsedConfirmedTransaction,
 } from "@solana/web3.js";
 import { useCluster, Cluster } from "../cluster";
 import * as Cache from "providers/cache";
 import { ActionType, FetchStatus } from "providers/cache";
 import { reportError } from "utils/sentry";
 
+const MAX_TRANSACTION_BATCH_SIZE = 30;
+
+type TransactionMap = Map<string, ParsedConfirmedTransaction>;
+
 type AccountHistory = {
   fetched: ConfirmedSignatureInfo[];
+  transactionMap?: TransactionMap;
```

Comment:
Why not use the transactions cache to store these?

Reply:
Let me think about that. That might be a better approach.